### DISCUSSION TITLE
Lighting in iModel

### DISCUSSION THREAD

**Discussion by glenworrall** [iTwin/itwinjs-core#5688] [Q&A] [answered]
Is it possible to insert lights into an iModel
ie If we use this lighting software
https://relux.com/en/relux-desktop.html?gad=1&gclid=EAIaIQobChMIiq-ugaTl_wIVweztCh3JeADqEAAYASAAEgJGavD_BwE
Can an iModel View handle lighting sources
Or if the lights are created in Microstation can they be loaded into an iModel ?

**Comment by MichelleJSAucier** [comment]
from lighting software ( Import / export numerous formats: Import (ies, ldt, ROLF, gbXML, ReCad, 3ds, dxf, dwg, xlsx, PDF, geolocations from GoogleMaps), Export (dwg, ReCad,, xlsx) )

https://developer.bentley.com/apis/synchronization/supported-formats/

In short you can export (lighting software) to dwg and import into iTwin, YES! please review list of third party applications that are able to be brought into the iTwin above.  Hope that this helps. 

This might help clarify the difference in iModel vs iTwin
 (https://communities.bentley.com/products/digital-twin-cloud-services/itwin-services/w/synchronization-wiki#:~:text=What%20is%20an%20iTwin%3F,is%20an%20infrastructure%20digital%20twin.&text=iModel%20connectors%20helps%20to%20transform,in%20to%20unified%20Imodel%20format. )

**Comment by MichelleJSAucier** [comment]
Or if the lights are created in Microstation can they be loaded into an iModel ?

the below video will show you how to attach reference files to MicroStation file. MicroStation can work with DGN, V7, and the DWG file formats. 

https://www.youtube.com/watch?v=DHNcPP4yl54

**Comment by pmconne** [comment]
@glenworrall there is a `BisCore:LightLocation` element representing a spatially-located light.
The only thing it specifies is whether or not the light is turned on or off - nothing I can see about its actual lighting properties.
@jffmarker or @abeesh could tell you which connectors import LightLocations into the iModel, and if (and if so, how) they preserve the lighting properties.
I can tell you that the display system currently does nothing with them.

**Reply by glenworrall** [reply]
Thanks @pmconne, yes it was more if there was a light imported ... can the display system know the luminance and create a light source similar to the "fire source" ...
https://www.itwinjs.org/sandboxes/iTwinPlatform/Fire%20Particle%20Effect
I see that the candle for example does not bounce ... but as I move away then the luminance becomes less visible
![image](https://github.com/iTwin/itwinjs-core/assets/17495289/9a152157-776b-444d-a653-a1870dfd85e9)
The idea being to increase the lumens of the light to determine which light source can be viewed from a specified distance.

**Reply by pmconne** [reply, marked as answer]
Your screenshot appears to be coming from a sample that demonstrates the use of particle effects, not lighting.
To reiterate: the iTwin.js display system currently does not support positional light sources.
Please describe the actual user problem you are trying to solve.

**Reply by jffmarker** [reply]
_RE: could tell you which connectors import LightLocations into the iModel, and if (and if so, how) they preserve the lighting properties._
I grepped for `lightlocations`, `lightlocation`, and `Lighting::Location`, and I only get hits in DGN and DWG conversion. DWG tries to set these: SetIntensity, SetColor, SetKelvin, SetType, SetBulbCount, SetSpot, SetLumens, SetShadowSamples on Lighting::Parameters. DGN sets these SetUserLabel, SetIntensity, SetColor, SetLumens, SetKelvin, SetShadowSamples, SetType, SetBulbCount, SetSpot.

**Reply by pmconne** [reply]
Thanks @jffmarker. Those all end up in the element's JSON properties. TypeScript API doesn't expose them nor do anything with them, but it sounds like they are getting imported.

**Reply by candcconsulting** [reply]
To respond to Paul's request
_Please describe the actual user problem you are trying to solve._
Some time ago you helped us create some functionality which allowed us to draw a light lozenge between the eye and a position in the iModel
ie here you can see I am 235m away from the signal
and the functionality draws the "light" lozenge from the signal to the camera eye
![image](https://github.com/iTwin/itwinjs-core/assets/19268085/74940ab3-16f8-4da8-87ae-f3204adf88bd)

The user would like to insert a light source so they can alter the luminance etc so it becomes more realistic.
Does that help ?

**Reply by pmconne** [reply]
> Does that help ?

It does - thank you.
However, at this time the display system will not actually produce any illumination from lights you insert into the iModel.
